SleepEngine/Engine/Utils/MacroUtils.h

```diff
@@ -24,6 +31,13 @@ classname& operator =(classname&&) = delete;
 FORBID_COPY(classname)\
 FORBID_MOVE(classname)
 
+#define SINGLETON_GETTER(class_)\
+static class_& Instance()\
+{\
+	static class_ instance;\
+	return instance;\
+}\
```

Comment:
Are you sure we need this underscore after 'class'? :D

Reply:
not to confuse with c++ class keyword,
i'm not sure wether macros variables can be named as keywords, but why do it anyway

Reply:
Ok